### PR TITLE
Upgrade caliban federation usage

### PR DESCRIPTION
**Pull request by paulpdaniels**

### CODE REVIEW

File: implementations/caliban/src/main/scala/ProductApi.scala [hunk 1]
```diff
@@ -64,28 +88,46 @@ object ProductApi extends GenericSchema[Has[ProductService]] {
...
+        ZQuery.serviceWithZIO[ProductService](_.getProductBySkuAndVariationId(sku, variation.id.id))
+
+    }
 
   case class Query(
-    product: IDArgs => URIO[Has[ProductService], Option[Product]]
+    product: IDArgs => URIO[ProductService, Option[Product]],
+    @GQLDeprecated("Use product query instead") deprecatedProduct: DeprecatedProductArgs => URIO[
+      ProductService,
+      DeprecatedProduct
```
Comment (dariuszkuc) [inline implementations/caliban/src/main/scala/ProductApi.scala:108]: I believe this should be `Option[DeprecatedProduct]`?

**Review by dariuszkuc — CHANGES_REQUESTED** [review]
Looks like your entity resolvers are not working/missing? Tests get null value vs expected object

```
* @key single › resolves single field @key on User
* @key multiple › resolves multiple field @key on DeprecatedProduct
* @key composite › resolves composite object @key on ProductResearch
```

`@requires` test also failed with null return value for `averageProductsCreatedPerYear`

**Comment by paulpdaniels** [comment]
@dariuszkuc comments addressed

**Review by dariuszkuc — APPROVED** [review]
Awesome! Thanks for the PR!
